### ISSUE TITLE
Model trained on custom dataset not detecting most objects

### ISSUE THREAD

**Issue by DarylWM** [AlexeyAB/darknet#2877]
Hi @AlexeyAB . I've learned a lot about YOLO by reading your answers here and the "How to improve object detection" section. I haven't managed to get it to detect many objects yet so I'm seeking your advice.

I'm training a YOLO model to detect peptide features in mass spectrometry data. I've converted the raw data into pseudo-colour images. I have 15,900 images in the training set, 885 images in the validation set, and 885 images in the test set (a 90:5:5% split), with images assigned to a set at random. Across the whole set, I have about 70,000 labelled objects from four classes.

After 5000 iterations the mAP hovers under 15-20% and never improves.

![image](https://user-images.githubusercontent.com/1016303/55764548-5ac2c000-5aaf-11e9-8305-399b751824a6.png)

As an example, for this image from the test set...

![image](https://user-images.githubusercontent.com/1016303/55764480-08819f00-5aaf-11e9-80bd-616baa9968db.png)

... this is the prediction:

![image](https://user-images.githubusercontent.com/1016303/55764504-2bac4e80-5aaf-11e9-847a-06116c8b60ae.png)

My model definition is a modified yolov3.cfg, with custom anchors for this dataset. The images are 910x910 but I had to make the YOLO input layer 864x864 to fit in my GTX1070s.
[yolov3-peptides.cfg.txt](https://github.com/AlexeyAB/darknet/files/3056666/yolov3-peptides.cfg.txt)

![image](https://user-images.githubusercontent.com/1016303/55764527-42eb3c00-5aaf-11e9-8c87-44ce7295b0ac.png)

Next on my list of things to try is a different model, like [5l](https://raw.githubusercontent.com/AlexeyAB/darknet/master/cfg/yolov3_5l.cfg) or [spp](https://raw.githubusercontent.com/AlexeyAB/darknet/master/cfg/yolov3-spp.cfg).

Do you have any suggestions?

**Comment by DarylWM** [comment]
Switching to yolov3-tiny_3l gave much better results, and it trains much faster.

![image](https://user-images.githubusercontent.com/1016303/55941339-62d45880-5c85-11e9-91d4-63959509bb56.png)

**Comment by AlexeyAB** [comment]
@DarylWM Hi,

Can you share your modified yolov3-tiny_3l.cfg?
This is strange that Tiny model gives better mAP than Full model.

**Comment by DarylWM** [comment]
Hi @AlexeyAB . From the cfg out of the box I changed the width and height, number of classes, filters, removed augmentation, and adjusted subdivisions.

[yolov3-tiny_3l-peptides.cfg.txt](https://github.com/AlexeyAB/darknet/files/3070861/yolov3-tiny_3l-peptides.cfg.txt)
